src/dkim/mod.rs

```diff
@@ -66,10 +64,9 @@ impl Default for Canonicalization {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[derive(Clone)]
 pub struct DomainKey {
-    pub(crate) v: Version,
-    pub(crate) p: PublicKey,
+    pub(crate) p: Arc<dyn VerifyingKey>,
```

Comment:
The effect here is that we'll have a nested `Arc` (both in `Txt` and inside `DomainKey`). I tried to avoid that but it wasn't very straightforward. I think it's probably not a big deal in the end -- the inner `Arc`'s refcount will just never change.

Reply:
For performance reasons I always try to avoid using dynamic dispatch and prefer static dispatch instead (either with enums or generics) as the code can be sometimes optimized by the compiler. Here there is also the extra cost of the nested ``Arc``. 
Probably the performance difference between ``dyn`` and an ``enum`` in this case is negligible since there are only very few options, but perhaps we can avoid ``Arc`` by manually implementing ``Clone`` for ``DomainKey`` and adding a clone method to the ``VeriyingKey`` trait? In any case clone won't ever be called since ``DomainKey`` is always used within ``Arc``.

Reply:
I think the consensus opinion of very experienced performance-focused Rust engineers is that Rust code tends to underuse dynamic dispatch (since the language kinds of discourages it using extra syntax etc). Although I've seen it matter in a hot loop (with many short/simple calls), in this case note that the only method on `VerifyingKey` is `verify()` which itself is expensive enough that a single dynamic dispatch won't matter.

I've changed the `Arc` to `Box`. It turns out `DomainKey` doesn't need to be `Clone` anyway (probably since it's always wrapped in an `Arc`), so that avoids doing boxed cloning through an explicit trait method.